girder_worker/plugins/docker/executor.py

```diff
@@ -118,6 +118,26 @@ def _docker_gc(tempdir):
     return subprocess.Popen(args=(script,), env=env)
 
 
+def validate_task_outputs(task_outputs):
+    """
+    This is called prior to fetching inputs to make sure the output specs are
+    valid. Outputs in docker mode can result in side effects, so it's best to
+    make sure the specs are valid prior to fetching.
+    """
+    for name, spec in task_outputs.iteritems():
+        if spec.get('target') == 'filepath':
+            if 'path' not in spec:
```

Comment:
@cdeepakroy this was discussed early on. The issue with a dictionary is that it is unordered. If you want to present inputs in a UI, they will be in an unordered jumble if the inputs are a dictionary. If you need to lookup inputs or outputs by name/id, you can always quickly make a map in `O(n)` time and then to quick lookups.

Also, I don't see input and output lists being long enough that `O(n)` lookups will cause any sort of overall performance issue.

So that's the rationale. If there was a good way to create convenience method that would help ease this I'd be all for it. There was some work on a Pythonic dict subclass for Girder Worker tasks that could have convenience methods like this - perhaps we could revisit that.